### PR TITLE
Apply aurelia-interface  support.

### PR DESCRIPTION
**Pull request by joelcoxokc**
This PR. Contains Several changes. 
No changes made to the original API... However, there are additions to the original API. 

1. Aurelia-Interface is using SASS/ SCSS. Added build-sass task. 
2. Cleaned up the build tasks to make gulp run quicker with less memory. 
3. Installed gulp-util along with a few other SASS Specific node_modules 
4. JSPM installed `joelcoxokc/aurelia-interface-sass-utils` a collection of utilities and variables used throughout the aurelia-interface. 
5. Added PRE-configuration for the `main.js` file
 - `@method: useBootstrap():` invoking this function will apply bootstrap's `modal` styles to the dialog. 
 - `@method: zIndex():` This method takes an `Object({dialog:Integer, overlay:Integer})` for initializing the base dialog and overly z-index's.

6. I removed any external element styling `e.i` the HTMLButtonElement. We were styling the button element... which could conflict with the end consumer's default button styles. 

7. I applied a few classes to the dialog and the overlay, when the `useBootstrap()` method is invoked.
8. Converted any extraneous styles to `flex-box`. I am using boubon.io, and an autopre-fixer to provide minimal `flex-box` support for IE 10+. There is currently no fallback support for IE 9. I will be making another PR with IE9 support.. without `flex-box`.

**NOTE** I left the original styles in the repo, for reference. 

-Joel**Cox**

### CODE REVIEW

File: build/tasks/serve.js [hunk 1]
```diff
@@ -10,10 +10,11 @@ gulp.task('serve', ['build'], function(done) {
   var bs = browserSync.create('Sample server');
 
   bs.init({
+    open: false,
```
Comment (PWKad) [inline build/tasks/serve.js]: Any reason to add this?

File: dist/amd/dialog-renderer.js [hunk 2]
```diff
@@ -126,14 +159,7 @@ define(['exports', 'aurelia-templating'], function (exports, _aureliaTemplating)
...
 
-        var vw = Math.max(document.documentElement.clientWidth, window.innerWidth || 0);
-        child.style.marginLeft = Math.max((vw - child.offsetWidth) / 2, 0) + 'px';
-
-        if (!settings.centerHorizontalOnly) {
-          var vh = Math.max(document.documentElement.clientHeight, window.innerHeight || 0);
-
-          child.style.marginTop = Math.max((vh - child.offsetHeight) / 2, 30) + 'px';
-        }
+        if (!settings.centerHorizontalOnly) {}
```
Comment (PWKad) [inline dist/amd/dialog-renderer.js]: Is this not needed anymore to center?

Comment (joelcoxokc) [inline dist/amd/dialog-renderer.js]: No. I am using flex. I have a way to do it in CSS without flex for IE9 as
well, I'll make the PR with the example tomorrow.

File: build/tasks/serve.js (hunk 1 above)
Comment (joelcoxokc) [inline build/tasks/serve.js]: Oh oops. Forgot to take that out. It kept opening new browser instances on me... And it go annoying lol

**Comment by EisenbergEffect** [comment]
@joelcoxokc Thanks for working on this! What would be your eta on adding the IE9 support? We don't need that for the rest of AI, but it would be nice to have for the dialog plugin if possible. Let me know.

**Comment by PWKad** [comment]
@joelcoxokc Few issues I've seen so far - 

1. Need to rebase this on current changes - not a big deal just matching up the configuration one way or another.  I can do this if you can finalize the changes.
2. Need to let settings stick (such as horizontal only centering)
3. Buttons aren't horizontally stacked any more by default.
4. Dist folders were `git rm`'d and needed to just not be included.

There may be a few others.  If you could fix the `dist` issue at least I can tackle the rest.  Thanks for your help!

**Comment by joelcoxokc** [comment]
Patrick. Ok I will take care of it. I thought I had fixed the dist.. But
just checked, and apparently I was wrong.

I also am going to have a meeting with Rob over the dialog tomorrow. Lets
wait till after that meeting.

**Comment by EisenbergEffect** [comment]
Let me know when you think this is ready.

**Comment by PWKad** [comment]
@Eisenbergeffect still waiting on the pr to be fixed to not delete dist folder from source control to fix the other issues.  @joelcoxokc

**Comment by EisenbergEffect** [comment]
@joelcoxokc I'm closing this for now since things have change significantly in both the dialog plugin and AI. After you get the AI refactoring done you mentioned, let's try to get a fresh PR against the dialog plugin to update it.

**Pull request closed without merging**